code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -71,11 +106,20 @@
 
 /obj/machinery/computer/camera_advanced/xenobio/attackby(obj/item/O, mob/user, params)
 	if(istype(O, /obj/item/reagent_containers/food/snacks/monkeycube))
-		monkeys++
-		to_chat(user, "<span class='notice'>You feed [O] to [src]. It now has [monkeys] monkey cubes stored.</span>")
-		user.drop_item()
-		qdel(O)
-		return
+		if(user.drop_item())
+			monkeys++
+			to_chat(user, "<span class='notice'>You feed [O] to [src]. It now has [monkeys] monkey cubes stored.</span>")
+			qdel(O)
+			return
+	else if (istype(O, /obj/item/slimepotion/slime))
+		var/replaced = FALSE
+		if(user.drop_item())
+			if(!QDELETED(current_potion))
+				current_potion.forceMove(drop_location())
+				replaced = TRUE
+			current_potion = O
```

Comment:
pretty sure this should be null

Reply:
That's pasted from the TG code, I was thinking the same thing cause it would be null if you haven't put in a potion yet. It works fine, so idk.

Reply:
should still be `null`.

Reply:
so like replace `if(!QDELTED(current_potion))` with `if(current_potion)`

Reply:
Yea that is supposed to be O, it replaces the current potion with the new one being inserted.

Reply:
oh, I just didn't look closely enough. it's `O` and not `0`.

Oops.